src/natcap/invest/wind_energy.py

```diff
@@ -2590,67 +2594,25 @@ def _calculate_distances_land_grid(base_point_vector_path, base_raster_path,
     # A list to hold the land to grid distances in order for each point
     # features 'L2G' field
     l2g_dist = []
-    # A list to hold the individual distance transform path's in order
+    # A list to hold the individual distance transform paths in order
     land_point_dist_raster_path_list = []
 
-    # Get the original layer definition which holds needed attribute values
-    base_layer_defn = base_point_layer.GetLayerDefn()
-    file_ext, driver_name = _get_file_ext_and_driver_name(
-        base_point_vector_path)
-    output_driver = ogr.GetDriverByName(driver_name)
-    single_feature_vector_path = os.path.join(
-        temp_dir, 'single_feature' + file_ext)
-    target_vector = output_driver.CreateDataSource(single_feature_vector_path)
-
-    # Create the new layer for target_vector using same name and
-    # geometry type from base_vector as well as spatial reference
-    target_layer = target_vector.CreateLayer(base_layer_defn.GetName(),
-                                             base_point_layer.GetSpatialRef(),
-                                             base_layer_defn.GetGeomType())
-
-    # Get the number of fields in original_layer
-    base_field_count = base_layer_defn.GetFieldCount()
-
-    # For every field, create a duplicate field and add it to the new
-    # shapefiles layer
-    for fld_index in range(base_field_count):
-        base_field = base_layer_defn.GetFieldDefn(fld_index)
-        target_field = ogr.FieldDefn(base_field.GetName(),
-                                     base_field.GetType())
-        # NOT setting the WIDTH or PRECISION because that seems to be
-        # unneeded and causes interesting OGR conflicts
-        target_layer.CreateField(target_field)
-
+    fid_field = base_point_layer.GetFIDColumn()
+    if not fid_field:
+        fid_field = 'FID'
     # Create a new shapefile with only one feature to burn onto a raster
     # in order to get the distance transform based on that one feature
     for feature_index, point_feature in enumerate(base_point_layer):
         # Get the point features land to grid value and add it to the list
-        field_index = point_feature.GetFieldIndex('L2G')
-        l2g_dist.append(float(point_feature.GetField(field_index)))
-
-        # Copy original_datasource's feature and set as new shapes feature
-        output_feature = ogr.Feature(feature_def=target_layer.GetLayerDefn())
-
-        # Since the original feature is of interest add its fields and
-        # Values to the new feature from the intersecting geometries
-        # The False in SetFrom() signifies that the fields must match
-        # exactly
-        output_feature.SetFrom(point_feature, False)
-        target_layer.CreateFeature(output_feature)
-        target_vector.SyncToDisk()
-        target_layer.DeleteFeature(point_feature.GetFID())
```

Comment:
This raised an exception from GDAL for some reason I'm not totally clear on. It seemed simpler to refactor this to avoid the situation - rather than creating a new vector for each individual point, we can filter.